src/Data/Text/Utf8.hs

```diff
@@ -65,11 +126,49 @@ unpackUtf8 (Text u8data offset length) =
   in
     go offset length
 
+lengthUtf8 :: Text -> CodeUnitIndex
+lengthUtf8 (Text _ _ !length) = CodeUnitIndex length
+
+-- TODO: Make this more readable once we have text-2.0.
+--
+-- >  off                 off+len
+-- >   │                     │
+-- >   ▼                     ▼
+-- > ──┬─┬─┬─┬─┬─┬─┬─┬─┬─┬─┬─┬─┬──
+-- >  A│B│C│D│E│F│G│H│I│J│K│L│M│N
+-- > ──┴─┴─┴─┴─┴─┴─┴─┴─┴─┴─┴─┴─┴──
+-- >       ▲           ▲
+-- >       │           │
+-- >  off+begin   off+begin+length
+--
+-- Visualizes:
+--
+-- > unsafeCutUtf8 2 6 "BCDEFGHIJKL" == ("BC", "JKL")
+unsafeCutUtf8 :: CodeUnitIndex -> CodeUnitIndex -> Text -> (Text, Text)
+unsafeCutUtf8 (CodeUnitIndex !begin) (CodeUnitIndex !length) (Text !u8data !off !len) =
+  ( Text u8data off begin
+  , Text u8data (off + begin + length) (len - begin - length)
+  )
```

Comment:
This is the biggest addition or at least the one I would be most worried about to break.

Reply:
Looks correct, but having some tests for it (especially for the cases where the input `off` is not 0) would help in establishing some confidence.